exonum/src/storage/memorydb.rs

```diff
@@ -51,18 +51,16 @@ impl Database for MemoryDB {
     }
 
     fn snapshot(&self) -> Box<Snapshot> {
-        Box::new(MemoryDB {
-            map: Arc::new(RwLock::new(self.map.read().unwrap().clone())),
-        })
+        Box::new(MemoryDB { map: Arc::clone(&self.map) })
     }
 
     fn merge(&mut self, patch: Patch) -> Result<()> {
+        let map = Arc::make_mut(&mut self.map);
```

Comment:
Is it OK here to (possibly) make a clone?

Reply:
I think yep: we are trading clones here for clones in `.snapshot` in the previous implementation. That is, if we do a `.clone` here, that means that refernce count is greater than one, and that means that we've done `.snapshot` before. That is, the total number of clones in this implementation will not be greater than in the original one.